index.php

```diff
@@ -160,96 +190,17 @@
 
 append_output_log("\nInput file: $basename\n");
 
-##### Figure out HTML version if needed, and generate output for user:
-$not_html5 = 0;
-if (str_ends_with($basename, ".html") || str_ends_with($basename, ".htm")) {
-    # Did we get HTML5? Look at the first line:
-    $tmpin = fopen($basename, "r");
-    if ($tmpin == false) {
-        throw new RuntimeException("Error opening file $basename for validation.");
-    }
-    $line = fgets($tmpin);
-    $regex = '/<\!DOCTYPE html/i';
-    if (! preg_match($regex, $line)) {
-        $not_html5 = 1;
-    }
-    # We also need to look for xhtml in the doctype line:
-    $regex = '/xhtml/i';
-    if (preg_match($regex, $line)) {
-        $not_html5 = 1;
-    }
-    fclose($tmpin);
-
-    # We got HTML5. Validate it:
-    if ($not_html5 == 0) {
-        print "<p style='color: green'>Validating your uploaded file...</p>";
-        append_output_log("Validating your uploaded file");
-        # TODO: what should we do if the validator fails? Log it? Error out?
-        system("$validator $basename >> $outfile 2>&1", $retval);
-        append_output_log("Validation complete");
-    }
-}
-
-if ($not_html5 || str_ends_with($basename, "xhtml")) {
-    echo <<<EOPARA
-    <p style='color: blue'>Your HTML does not seem to be HTML version
-    5. Be sure to run the appropriate validator, probably the W3C's
-    <a href="https://validator.w3.org">legacy HTML &amp; CSS
-    validator</a>.</p>
-    EOPARA;
-
-    append_output_log("--- HTML validation not done");
-    append_output_log("HTML validation not done since your source is not HTML5.");
-    append_output_log("You must use a validator such as https://validator.w3.org in addition to ebookmaker");
-    append_output_log("---");
-}
-
-// Options to ebookmaker:
-$opts = [
-    "--make", "txt.utf-8",
-    "--make", "epub",
-    "--make", "epub3",
-    "--make", "kf8",
-    "--make", "kindle",
-    "--make", "html",
-    "--validate",
-    "--max-depth", 3,
-    "--output-dir", "$tmpdir/$tmpsubdir",
-];
-
-# required argument
-$opts[] = "--ebook";
-if (strlen($_REQUEST['myebook'])) {
-    $opts[] = escapeshellarg($_REQUEST['myebook']);
-} else {
-    $opts[] = 10001; # Required
-}
-if (strlen($_REQUEST['mytitle'])) {
-    $opts[] = "--title";
-    $opts[] = escapeshellarg($_REQUEST['mytitle']);
-}
-if (strlen($_REQUEST['myauthor'])) {
-    $opts[] = "--author";
-    $opts[] = escapeshellarg($_REQUEST['myauthor']);
-}
-
-$gopts = join(" ", $opts);
+validate_html($basename);
```

Comment:
@windymilla - do we want to make validating the HTML optional for Guiguts?

Reply:
We offer Nu validator submission as an integration in Guiguts already, and our local Ebookmaker integration skips the validation step. But it could be convenient to have both done together since we are going round trip to the server already, and users of online ebookmaker currently expect that invalid input will result in errors.

So on net, I would lean toward not making it optional.

Reply:
I agree - let's not make it optional. I think we want the results the user sees in GG2 to be as close as possible to the results they would see online.

Reply:
I don't think this is all the same thing.

If uploaded an HTML file, or a zip file with an HTML file, `ebookmaker-web` will run that directly through the Nu validator _before_ putting into ebookmaker.

Separately, ebookmaker uses the `--validate` parameter to `validate epub and html through epubcheck/nu` according to the help message.

Do we want to make the first validation optional? I think yes and I've added an API field to optionally turn it off with `validate_html=False` if the API user wanted to, but it defaults to `True`.